src/Microsoft.OpenApi/Models/OpenApiSchema.cs

```diff
@@ -437,7 +437,10 @@ private void SerializeInternal(IOpenApiWriter writer, OpenApiSpecVersion version
             });
 
             // enum
-            writer.WriteOptionalCollection(OpenApiConstants.Enum, Enum, (nodeWriter, s) => nodeWriter.WriteAny(s));
+            var enumValue = !string.IsNullOrEmpty(Const) && version != OpenApiSpecVersion.OpenApi3_1
```

Comment:
```suggestion
            var enumValue = !string.IsNullOrEmpty(Const) && version < OpenApiSpecVersion.OpenApi3_1
```
to be future proof with 3.2